wgpu-core/src/device/mod.rs

```diff
@@ -1478,7 +1480,51 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let device = &device_guard[texture.device_id.value];
 
         let view_kind = match desc.dimension {
-            Some(dim) => conv::map_texture_view_dimension(dim),
+            Some(dim) => {
+                use hal::image::Kind;
+
+                let valid_view_dim = match texture.dimension {
```

Comment:
can we rewrite this in a simpler way without `matches!`, like:
```rust
let texture_dim = match dim {
  TextureViewDimension::D1 => TextureDimension::D1,
  ... => TextureDimension::D2,
  TextureViewDimension::D3 => TextureDimension::D3,
};
if texture.dimension != texture_dim {
  return Err(..);
}
```
We could even put this conversion into `wgt` if it's useful for public

Reply:
Seems like it could be, I'll add it there.